client/ayon_core/tools/attribute_defs/widgets.py

```diff
@@ -313,14 +313,29 @@ def _ui_init(self):
         self.main_layout.addWidget(multisel_widget, 0)
 
     def eventFilter(self, obj, event):
-        if (
+        if event.type() == QtCore.QEvent.Type.ContextMenu:
+            QtCore.QTimer.singleShot(0, self._on_menu_timeout)
+            super(NumberAttrWidget, self).contextMenuEvent(event)
+        elif (
             self._multivalue
             and obj is self._input_widget
             and event.type() == QtCore.QEvent.FocusOut
         ):
             self._set_multiselection_visible(True)
         return False
 
+    def _on_menu_timeout(self):
+        menu = self.findChild(QtWidgets.QMenu, "qt_edit_menu")
+        if menu is not None:
+            first_action = menu.actionAt(QtCore.QPoint())
+            action = QtWidgets.QAction(menu)
+            action.setText("Reset to default")
+            action.triggered.connect(self.reset_to_default_requested)
+            menu.insertAction(first_action, action)
+
+    def reset_to_default_requested(self):
```

Comment:
The method name is not ideal. It should be something like `set_to_default_value`, or private method for signal handling and in that case `_on_default_value_requested`.